src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1060,12 +1061,11 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
   }
 
   private def withWorkspaceContext(workspaceName: WorkspaceName, writeLock: Boolean, txn: RawlsTransaction)(op: (WorkspaceContext) => Future[PerRequestMessage]) = {
-    containerDAO.workspaceDAO.loadContext(workspaceName, txn) match {
-      case None => Future.successful(RequestComplete(ErrorReport(StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))))
-      case Some(workspaceContext) =>
-        txn.withLock(workspaceContext.workspaceVertex, writeLock) {
-          op(workspaceContext)
-        }
+    txn.withFutureLock(workspaceName, writeLock) {
```

Comment:
Now the lock takes a `WorkspaceName` rather than a `Vertex`, we can "lock the workspace" even before we've loaded the context.

Also note that this is the Future-y version because WorkspaceService calls are Future-y.